pkg/sources/job_progress.go

```diff
@@ -94,11 +94,11 @@ type ChunkError struct {
 }
 
 func (f ChunkError) Error() string {
-	unit := "<nil>"
+	unit, kind := "<nil>", SourceUnitKind("unit")
```

Comment:
Would it be helpful to use a special "nil unit" kind here? Or would that be more trouble than it's worth?

Reply:
Hm.. I don't see a strong need to add that currently, so I'm inclined to leave it as is. I'll keep that idea in the back of my head though.